### PR TITLE
Add authParams object for specifying additional parameters...

### PR DESCRIPTION
**Pull request by oetkb**
to be added to the OAuth authz request. 

Allows specifying scope, custom parameters (for Access Manager), etc.

### CODE REVIEW

File: src/main/js/oauth.js [hunk 1]
```diff
@@ -292,6 +292,15 @@ gromit.oAuthAuthenticate = function(/*String*/ url, /*function*/ callback, /*boo
     url += '?response_type=token';
     url += '&redirect_uri=' + gromit.getCurrentUrl() + 'oauth.html';
     url += '&client_id=' + gromit.ClientId;
+    
+    if (typeof gromit.AuthParams !== 'undefined') {
```
Comment (zgrossbart) [inline src/main/js/oauth.js]: This isn't a great way to check if something is defined in JavaScript.  You can just say `if (gromit.AuthParams)`

File: src/main/js/oauth.js [hunk 2]
```diff
@@ -292,6 +292,15 @@ gromit.oAuthAuthenticate = function(/*String*/ url, /*function*/ callback, /*boo
...
+        for (var i = 0; i < gromit.AuthParams.length; i++) {
+            url += '&' + gromit.AuthParams[i].name;
```
Comment (zgrossbart) [inline src/main/js/oauth.js]: You probably don't want to add the parameter to the URL if there is a name and no value.

File: src/main/js/oauth.js [hunk 3]
```diff
@@ -292,6 +292,15 @@ gromit.oAuthAuthenticate = function(/*String*/ url, /*function*/ callback, /*boo
...
+            if (gromit.AuthParams[i].value !== 'null') {
+                url += '=' + gromit.AuthParams[i].value;
```
Comment (zgrossbart) [inline src/main/js/oauth.js]: Do we need to URL encode these values?

**Review by zgrossbart — CHANGES_REQUESTED** [review]

File: src/main/js/oauth.js (hunk 1 above)
Comment (oetkb) [reply src/main/js/oauth.js]: Yeah, whatever. I'm not exactly a JS wizard.

File: src/main/js/oauth.js (hunk 2 above)
Comment (oetkb) [reply src/main/js/oauth.js]: A valueless parameter is valid. Rarely used, but may as well allow it.

File: src/main/js/oauth.js (hunk 3 above)
Comment (oetkb) [reply src/main/js/oauth.js]: No more nor less than the redirect_uri, client_id or anything else that could have reserved characters. I figured that the supplier of data (i.e., the server side) could handle that if necessary.

**Review by zgrossbart — APPROVED** [review]
Looks great.  Thanks.

**Comment by oetkb** [comment]
Hey Zack, can you merge this pull request or do I need to wait for Irina to review?

**Comment by zgrossbart** [comment]
I'm sorry.  I didn't realize you were waiting for me.  You're all set.

**Comment by oetkb** [comment]
No worries. Thanks for taking care of it.

**Pull request merged**